unison-cli/src/Unison/Share/SyncV2.hs

```diff
@@ -116,6 +117,9 @@ syncFromFile ::
   Cli (Either (SyncError SyncV2.PullError) CausalHash)
 syncFromFile shouldValidate syncFilePath = do
   Cli.Env {codebase} <- ask
+  -- Every insert into SQLite checks the temp entity tables, but syncv2 doesn't actually use them, so it's faster
+  -- if we clear them out before starting a sync.
+  Cli.runTransaction Q.clearTempEntityTables
```

Comment:
The temp entity table should be empty except if a past sync failed and was never completed, right?

Reply:
Yup; but it's also never otherwise cleared, so if you accidentally `clone @unison/base/main` instead of `lib.install @unison/base` or something like that it'll leave junk in there basically forever.